content/blog/en/version-6-dojo.md

```diff
@@ -0,0 +1,295 @@
+
+We are really looking forward to seeing the fun and innovative middlewares from the Dojo community, and, as always, please let us know any feedback that you might have!
+
+## Config free Custom Elements
+
+Including Web Components as a first class citizen in Dojo is something that we’ve always been passionate about and now compiling your Dojo widgets to Custom Elements is _even_ easier. In Dojo 6 there is no configuration required, other than defining the widget(s) in the `.dojorc` to instruct the `@dojo/cli-build-widget` to compile your widgets to custom elements. We think this is a significant improvement to current tooling that requires widgets to be explicitly configured with the custom element details. The approach prior to Dojo version 6 required additional development effort, foresight, and created an additional maintenance burden of keeping the configuration up to date with changes to widget properties. This can be seen below with the custom element configuration for an example widget.
+
+Configuration required prior to version 6 for compiling a Dojo widget to a custom element:
+
+> .dojorc
+```json
+{
+  "build-widget": {
+    "widgets": [
+      "src/MyWidget"
+    ]
+  }
+}
+```
+
+> src/MyWidget
+```tsx
+import { WidgetBase } from '@dojo/framework/core/WidgetBase';
+import { customElement } from '@dojo/framework/core/decorators/customElement';
+import { CustomElementChildType } from '@dojo/framework/core/registerCustomElement';
+
+interface MyWidgetProperties {
+  value: string;
+  disabled: boolean;
+  onInput: (input: value) => void;
+}
+
+@customElement<MyWidgetProperties>({
+	tag: 'dojo-my-widget',
+	childType: CustomElementChildType.TEXT,
+	properties: ['disabled'],
+	attributes: ['value'],
+	events: [
+		'onInput'
+	]
+})
+class MyWidget extends WidgetBase<MyWidgetProperties> {
+  protected render() {
+    // rendering
+  }
+}
+```
+
+In Dojo version 6, this is substantially simplified. Now the Dojo build tool's intelligent configuration of custom elements only requires the `.dojorc` entry and will automatically include new, changed or removed properties!
+
+> .dojorc
+```json
+{
+  "build-widget": {
+    "widgets": [
+      "src/MyWidget"
+    ]
+  }
+}
+```
+
+## BTR and Dojo Block Improvements
+
+In version 5 of Dojo, we announced Dojo Blocks, a feature leveraging build time rendering (BTR) that brought the world of static site generation to Dojo. Since then we have been working on improving the experience including more intelligent block bundling, dynamic path registration and a full static mode. Building static and progressive websites has never been easier with Dojo.
+
+Review the [example static blog site with Dojo on CodeSandbox](https://codesandbox.io/s/my-first-blog-bywnn) to learn more about BTR and Dojo Block improvements.
+
```

Comment:
With Dojo 6, BTR has gotten even smarter. During the build, BTR will figure out what scripts and CSS assets were used and add them to the rendered output as preload tags. Downloading these important files right away, and before they are needed by the page, results in faster page performance. Additionally, assets that are not used by the current page, but are used by other pages, are added to the page as "prefetch", which instructs the browser to download these files when it's idle. Once the user interacts and chooses one of these other pages, the assets for the page will have already been loaded by the browser's prefetch. This results in a substantial load-time improvement.

Reply:
@rorticus Added. Thanks!